Add Subaru integration (#13493)
home-assistant/home-assistant.github.io: source/_integrations/subaru.markdown

---
title: Subaru 
description: Instructions on how to integrate Subaru car into Home Assistant.
ha_category:
  - Car
  - Sensor

> ha_release: 0.117

Reviewer: ```suggestion
ha_release: 2021.3
```

ha_iot_class: Cloud Polling
ha_config_flow: true
ha_codeowners:
  - '@G-Two'
ha_domain: subaru
---

The `Subaru` integration connects to the [MySubaru](https://www.mysubuaru.com) cloud service to provide vehicle sensor information as well as the capability to remotely actuate door locks. This integration requires an active subscription to the MySubaru service.

This integration provides the following platforms:

- Sensors - Such as outside temperature, average fuel consumption, tire pressure, odometer, estimated range, and EV information (battery level, range, charging rate).

> ## Configuration
> 
> The Subaru integration offers configuration through the Home Assistant UI:
>     
> **Configuration** -> **Integrations** -> **Add** -> **Subaru**
> 
> When prompted, enter the following configuration parameters:
> 
> - **Email Address:** The email address associated with your MySubaru account.
> - **Password:** The password associated with your MySubaru account.
> - **PIN:** The PIN associated with your MySubaru account. 
>   - Note: If your account includes multiple vehicles, the same PIN will need to be used for all vehicles.
> 
> First-time validation of your credentials may take up to 30 seconds. 

Reviewer: ```suggestion
{% include integrations/config_flow.md %}
```

You'll need to rebase for that to work.

---
title: Subaru 
description: Instructions on how to integrate Tesla car into Home Assistant.
ha_category:
  - Car
  - Binary Sensor
  - Climate
  - Presence Detection
  - Lock
  - Sensor
  - Switch
ha_release: 0.111
ha_iot_class: Cloud Polling
ha_config_flow: true
ha_codeowners:
  - '@G-Two'
ha_domain: subaru
---

The `Subaru` integration offers integration with the [Subaru](https://www.mysubuaru.com) cloud service and provides presence detection as well as sensors such as charger state and temperature.

This integration provides the following platforms:

> - Binary sensors - EV charger connection status.
> - Sensors - such as Outside temperature, odometer, estimated range, and EV information (battery level, range, charging rate).
> - Device tracker - obtain location of your car.
> - Lock - Lock and Unlock Doors.
> - Climate - Enables control of HVAC and Remote Engine Start (turn on/off, set target temperature).
> - Switch - Remote actuate Horn, Lights, request Location, and start EV charging.

Reviewer 2: ```suggestion
- Binary sensors - EV charger connection status.
- Sensors - Such as Outside temperature, odometer, estimated range, and EV information (battery level, range, charging rate).
- Device tracker - Obtains the location of your car.
- Lock - Lock and Unlock Doors.
- Climate - Enables control of HVAC and Remote Engine Start (turn on/off, set target temperature).
- Switch - Remote actuate Horn, Lights, request Location, and start EV charging.
```

## Configuration

### Home Assistant UI
  * The Subaru integration offers configuration through the Home Assistant UI:
    
    **Configuration** -> **Integrations** -> **Add** -> **Subaru**

> ### YAML

Reviewer 2: Since this is a new integration, that interacts with a device or service, it should not supply YAML based configurations.

See ADR-0010: <https://github.com/home-assistant/architecture/blob/master/adr/0010-integration-configuration.md>

Author: YAML references removed.  Thanks for pointing me towards the ADRs.
